Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -1807,21 +1807,53 @@ private bool DoesActivationMatchFocus(FocusEventData eventData)
 
         void IMixedRealityFocusChangedHandler.OnFocusChanged(FocusEventData eventData)
         {
-            if (activation == BoundingBoxActivationType.ActivateManually || activation == BoundingBoxActivationType.ActivateOnStart)
+            if (activation != BoundingBoxActivationType.ActivateManually && activation != BoundingBoxActivationType.ActivateOnStart)
             {
-                return;
+                if (DoesActivationMatchFocus(eventData))
+                {
+                    bool handInProximity = eventData.NewFocusedObject != null && eventData.NewFocusedObject.transform.IsChildOf(transform);
+                    if (handInProximity == wireframeOnly)
+                    {
+                        wireframeOnly = !handInProximity;
+                        ResetHandleVisibility();
+                    }
+                }
             }
 
-            if (!DoesActivationMatchFocus(eventData))
+            if (eventData.NewFocusedObject != null && eventData.NewFocusedObject.transform.IsChildOf(transform))
             {
-                return;
-            }
+                GameObject grabbedHandle = eventData.Pointer.Result.CurrentPointerTarget;
+                if (grabbedHandle)
+                {
+                    currentHandleType = GetHandleType(grabbedHandle.transform);
 
-            bool handInProximity = eventData.NewFocusedObject != null && eventData.NewFocusedObject.transform.IsChildOf(transform);
-            if (handInProximity == wireframeOnly)
+                    var bc = eventData.Pointer.BaseCursor as BaseCursor;
+                    if (bc != null)
+                    {
+                        bc.handleRigTransform = rigRoot.transform;
```

Comment:
Setting handleRig from manipulationhandler feels like it's creating a bit of a coupling between BaseCursor and ManipulationHandler. You need to know that in order for the resize Rotate or Scale cursor actions to work, you also need to set handleRigTransform, and that specifically it needs to match the bounding box rigRoot transform (or at least have the same shape).

Is there a more general way we could do this?


Imagine you were an external developer trying to figure out how to make the cursor show the diagonal resize cursor on your own custom asset on hover. It would require some sleuthing into our code to realize that you need to set handleRigTransform, and further that it needs to match the bounding box.